spec/support/ldap_shared_examples.rb

```diff
@@ -0,0 +1,91 @@
+shared_context 'mocked ldap' do |returns:|
+  let(:entry_users) { send(returns) }
+  let(:expected_entries) {
+    entry_users.map { |test_user|
+      expected_entry = Net::LDAP::Entry.new
+      expected_entry[:uid] = test_user[:username]
+      expected_entry[:givenName] = test_user[:first_name]
+      expected_entry[:sn] = test_user[:last_name]
+      expected_entry[:mail] = test_user[:email]
+      expected_entry[:displayName] = test_user[:display_name]
+      expected_entry
+    }
+  }
+  before do
+    allow_any_instance_of(Net::LDAP).to receive(:search).and_return(expected_entries)
```

Comment:
Cool! I'm sure there are other places in our code that could use this technique.